src/main.rs

```diff
@@ -441,40 +439,41 @@ fn parse_numeric_range_opts(
     min_value: u8,
     max_value: u8,
 ) -> Result<HashSet<u8>, String> {
-    let one_item = Regex::new(format!(r"^[{}-{}]$", min_value, max_value).as_ref()).unwrap();
-    let multiple_items = Regex::new(
-        format!(
-            r"^([{}-{}])(,|-)([{}-{}])$",
-            min_value, max_value, min_value, max_value
-        ).as_ref(),
-    ).unwrap();
+    let error_message = "Not a valid input";
     let mut items = HashSet::new();
 
-    if one_item.is_match(input) {
-        items.insert(input.parse::<u8>().unwrap());
-        return Ok(items);
+    // one value
+    if let Some(one_value) = input.parse::<u8>().ok() {
+        if (min_value <= one_value) && (one_value <= max_value) {
+            items.insert(one_value);
+            return Ok(items);
+        }
     }
 
-    if let Some(captures) = multiple_items.captures(input) {
-        let first = captures[1].parse::<u8>().unwrap();
-        let second = captures[3].parse::<u8>().unwrap();
-        if first >= second {
-            return Err("Not a valid input".to_owned());
+    // a range ("A-B")
+    let range_values = input.split('-').collect::<Vec<&str>>();
+    if range_values.len() == 2 {
+        let first_opt = range_values[0].parse::<u8>().ok();
+        let second_opt = range_values[1].parse::<u8>().ok();
+        if let (Some(first), Some(second)) = (first_opt, second_opt) {
+            if min_value <= first && first < second && second <= max_value {
+                for i in first..second + 1 {
+                    items.insert(i);
+                }
+                return Ok(items);
+            }
         }
+    }
```

Comment:
This will silently fail and use the default option if someone puts in e.g. `1-` or `1-2-3` or `1-A`.